src/oscar/templatetags/category_tags.py

```diff
@@ -1,12 +1,73 @@
+from six import with_metaclass
 from django import template
-
 from oscar.core.loading import get_model
 
 register = template.Library()
-Category = get_model('catalogue', 'category')
+Category = get_model("catalogue", "category")
+
+
+class PassThrough(object):
+    def __init__(self, name):
+        self.name = name
+
+    def __get__(self, obj, objtype):
+        if obj is None:
+            return self
+
+        return getattr(obj.category, self.name)
+
+
+class CategoryFieldPassThroughMetaClass(type):
+    """
+    Add accessors for category fields to whichever class is of this type.
+    """
+    def __new__(cls, name, bases, attrs):
+
+        field_accessors = {}
+        for field in Category._meta.get_fields():
+            name = field.name
+            field_accessors[name] = PassThrough(name)
+
+        # attrs win of silly field accessors
+        field_accessors.update(attrs)
+        return type.__new__(cls, name, bases, field_accessors)
+
+
+class CheapCategoryInfo(with_metaclass(CategoryFieldPassThroughMetaClass, dict)):
+    """
+    Wrapper class for Category.
+
+    Besides allowing inclusion of extra info, useful while rendering a template,
+    this class hides any expensive properties people should not use by accident
+    in templates.
+
+    This replaces both the node as the info object returned by the ``category_tree``
+    templatetag, so it mimics a tuple of 2 items (which are the same) for
+    backwards compatibility.
+    """
+
+    def __init__(self, category, **info):
+        super(CheapCategoryInfo, self).__init__(info)
```

Comment:
Please use `super()` for consistency across the project. We don't support Python 2.